tests/tekton-resources/tasks/setup/eks/awscli-mng.yaml

```diff
@@ -72,33 +82,100 @@ spec:
-          --node-role $NODE_ROLE_ARN \
-          --launch-template name=$launch_template_name\
-          --region $(params.region) \
-          --instance-types $EC2_INSTANCES \
-          --scaling-config minSize=$(params.min-nodes),maxSize=$2,desiredSize=$2 \
-          --subnets $NG_SUBNETS $TAINTS_FLAG
+        # if no unmanaged nodegroup cfn template is provided, assume we want managed nodegroups
+        if [ "$(params.unmanaged-nodegroup-cfn-url)" = "" ]; then
+          CREATED_NODEGROUP=$(aws eks $ENDPOINT_FLAG --region $(params.region) list-nodegroups --cluster-name $(params.cluster-name)  --query 'nodegroups[?@==`'$node_group_name'`]' --output text)
+          if [ "$CREATED_NODEGROUP" == "" ]; then
+            aws eks $ENDPOINT_FLAG create-nodegroup \
+            --cluster-name $(params.cluster-name) \
+            --nodegroup-name $node_group_name \
+            --node-role $NODE_ROLE_ARN \
+            --launch-template name=$(params.launch-template-name) \
+            --region $(params.region) \
+            --instance-types $EC2_INSTANCES \
+            --scaling-config minSize=$(params.min-nodes),maxSize=$2,desiredSize=$2 \
+            --subnets $NG_SUBNETS $TAINTS_FLAG
+          fi
+          echo "CREATED_NODEGROUP=$node_group_name"
+          while [[ "$(aws eks $ENDPOINT_FLAG --region $(params.region) describe-nodegroup --cluster-name $(params.cluster-name) --nodegroup-name $node_group_name --query nodegroup.status --output text)" == "CREATING" ]]
+          do
+            echo "$node_group_name is "CREATING" at $(date)"
+            sleep 2
+          done
+          # TODO: do this for unmanaged nodes as well
+          # right now we don't have an appropriate label to filter on for unmanaged nodes
+          while true; do
+            ready_node=$(kubectl get nodes -l eks.amazonaws.com/nodegroup=$node_group_name --no-headers 2>/dev/null | grep -w Ready | wc -l)
+            echo "ready-nodes=$ready_node out of $2, for nodegroup: $node_group_name"
+            if [[ "$ready_node" -eq $2 ]]; then break; fi
+            sleep 5
+          done
+        else
+          echo "Creating or updating aws-auth ConfigMap..."
+          if ! kubectl get configmap aws-auth -n kube-system &> /dev/null; then
+            # Download the official template as shown in the Amazon EKS User Guide:
+            # https://docs.aws.amazon.com/eks/latest/userguide/auth-configmap.html#aws-auth-configmap
+            curl -O https://s3.us-west-2.amazonaws.com/amazon-eks/cloudformation/2020-10-29/aws-auth-cm.yaml
+
+            # Replace the placeholder with our role ARN and apply the configmap
+            sed -i.bak -e "s|<ARN of instance role (not instance profile)>|${NODE_ROLE_ARN}|" aws-auth-cm.yaml
+
+            kubectl apply -f aws-auth-cm.yaml
+            echo "Created aws-auth ConfigMap"
+            # Wait for the config map to be ready
+            echo "Verifying aws-auth ConfigMap..."
```

Comment:
If `kubectl apply` exit with 0, then the aws-auth CM should have been applied successfully.
Is there a scenario that is not covered by that?